pkg/frontend/export.go

```diff
@@ -137,54 +125,61 @@ func initExportFileParam(ep *ExportConfig, mrs *MysqlResultSet) {
 }
 
 var openNewFile = func(ctx context.Context, ep *ExportConfig, mrs *MysqlResultSet) error {
-	lineSize := ep.LineSize
 	var err error
+	var filePath string
 	ep.CurFileSize = 0
-	if !ep.UseFileService {
-		var filePath string
-		if len(ep.userConfig.StageFilePath) != 0 {
-			filePath = getExportFilePath(ep.userConfig.StageFilePath, ep.FileCnt)
-		} else {
-			filePath = getExportFilePath(ep.userConfig.FilePath, ep.FileCnt)
-		}
-		ep.File, err = OpenFile(filePath, os.O_RDWR|os.O_EXCL|os.O_CREATE, 0o666)
+
+	ep.AsyncReader, ep.AsyncWriter = io.Pipe()
+	if len(ep.userConfig.StageFilePath) != 0 {
+		filePath = getExportFilePath(ep.userConfig.StageFilePath, ep.FileCnt)
+	} else {
+		filePath = getExportFilePath(ep.userConfig.FilePath, ep.FileCnt)
+	}
+
+	// fileservice is determined from filepath
+	// if filepath is SHARED:/path, return getGlobalPu().FileService, filepath
+	// otherwise, return fileservice.GetForETL(ctx, nil, filepath)
+	fspath, err := fileservice.ParsePath(filePath)
+	if err != nil {
+		return err
+	}
+
+	var readPath string
+	if fspath.Service == defines.SharedFileServiceName {
```

Comment:
I just check SharedFileServiceName (SHARED) to use default file service.  Shall I also check LocalFileServiceName (LOCAL) too?

Reply:
using SHARED is the right choice always, I think